### PR TITLE
docker: Pull llvm-objcopy in cilium-builder

### PR DESCRIPTION
**Pull request by pchaigno**
This pull request pulls in the `llvm-objcopy` binary from the cilium-llvm image to the cilium-builder image. This binary is required to remove sections from object files loaded with bpftool in the K8sVerifier test.

Other references to cilium-llvm are also updated in this PR.

### CODE REVIEW

**Comment by pchaigno** [comment]
test-only --focus="K8sVerifier" --kernel_version=net-next

**Comment by pchaigno** [comment]
test-me-please

File: Dockerfile.builder [hunk 1]
```diff
@@ -33,6 +33,11 @@ RUN apt-get update && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+#
+# Retrieve llvm-objcopy binary
+#
+COPY --from=docker.io/cilium/cilium-llvm:cae23fe2f43497ae268bd8ec186930bc5f32afac /usr/local/bin/llvm-objcopy /bin/
```
Comment (twpayne) [inline Dockerfile.builder]: nit: is it possible to make the image an `ARG` here as done in `images/cilium/Dockerfile` below?

Specifically:
* Add `ARG CILIUM_LLVM_IMAGE=docker.io/cilium/cilium-llvm:cae23fe2f43497ae268bd8ec186930bc5f32afac` towards the top of the file.
* Write this line as `COPY --from=$CILIUM_LLVM_IMAGE /usr/local/bin/llvm-objcopy /bin/`

This avoids having a hardcoded magic number (i.e. the SHA256 hash) in the code and both adds meaning to the number and makes it easier to change in the future.

**Review by twpayne — APPROVED** [review]
LGTM, one small nit.

File: Dockerfile.builder (hunk 1 above)
Comment (pchaigno) [reply Dockerfile.builder]: Oh, nice! I didn't know about that.

Though, Docker doesn't seem to like it in this case (no pun intended):
```
Step 11/12 : COPY --from=${CILIUM_LLVM_IMAGE} /usr/local/bin/llvm-objcopy /bin/
invalid from flag value ${CILIUM_LLVM_IMAGE}: invalid reference format: repository name must be lowercase
```
I'm guessing that's why we don't do that for any other image we have.

**Review by aanm — CHANGES_REQUESTED** [review]
Since `Dockerfile.builder` was modified, `Dockerfile.runtime` also needs to be updated.
Since `Dockerfile.runtime` is modified, `Dockerfile` also needs to be updated.

**Comment by pchaigno** [comment]
> Since `Dockerfile.builder` was modified, `Dockerfile.runtime` also needs to be updated.
> Since `Dockerfile.runtime` is modified, `Dockerfile` also needs to be updated.

I couldn't find any reference to cilium-builder in `contrib/packaging/docker/Dockerfile.runtime`. What should be updated?

How can I update `Dockerfile` with the new cilium-builder reference before this PR is merged and the new cilium-builder image built?

**Comment by pchaigno** [comment]
retest-net-next

**Comment by pchaigno** [comment]
test-me-please

**Comment by pchaigno** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Review by jibi — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by pchaigno** [comment]
We need to include `llvm-objcopy` in `cilium-builder` for the K8sVerifier test which we're backporting to v1.8 and v1.9 (cf. https://github.com/cilium/cilium/pull/14797 and https://github.com/cilium/cilium/pull/14798).

**Pull request merged**